quarkus/config-api/src/main/java/org/keycloak/config/EventOptions.java

```diff
@@ -0,0 +1,21 @@
+package org.keycloak.config;
+
+public class EventOptions {
+
+    public static final Option<Boolean> USER_EVENT_METRICS_ENABLED = new OptionBuilder<>("user-event-metrics-enabled", Boolean.class)
+            .category(OptionCategory.EVENTS)
+            .description("Create metrics based on user events.")
+            .buildTime(false)
+            .defaultValue(Boolean.FALSE)
+            .build();
+
+    public static final Option<String> USER_EVENT_METRICS_TAGS = new OptionBuilder<>("user-event-metrics-tags", String.class)
+            .category(OptionCategory.EVENTS)
+            .description("Comma-separated list of tags to be collected for events. Reduce the tags to avoid a high metrics cardinality if you use a lot of client, realms or IDPs")
+            .buildTime(false)
+            .defaultValue("realm,clientId,idp")
```

Comment:
What does `idp` refer to? Is that login via external IdP?

Creating dimensions for clients by default is not great IMO. Say there's 5 realms each with 10K clients, and you have something like 20+ (not sure what events are counted) events, that would add 1000000 metrics, which would use a lot of memory + would make metrics endpoint very slow (probably even fail).

I would honestly drop the clientId completely, as it shouldn't be there. As an example you don't have metrics on a per-user basis, so makes no sense to have them on a per-client basis.

Reply:
@stianst - it is the IDP in the events, so yes, the external IDP. 

I agree that there are installations with a lot of clients, and they would then disable the dimension clients. That's why there is this option to disable them, as well as the idp. 

There was also the idea to limit the events that are used to generate metrics. We can still add this if there is the feedback around that.

Reply:
Does it log all events right now, that's something like 150+ events. Seems like we should rather just log some specific list of events?

I'm not convinced about the clients - I can see it could be useful to be able to identify noisy clients, but also don't like that we know it's going to be an issue with a certain amount of clients, and can only support this for smaller amounts of clients. It's basically a feature that is only available to small deployments, but not large deployments.

Reply:
It will only be user events, which will be fewer. I'll come up with good default list with common events even for the first iteration then.